wagtail/wagtailcore/fields.py

```diff
@@ -1,18 +1,69 @@
 from __future__ import absolute_import, unicode_literals
 
 import json
+import sys
 
-from django.db import models
 from django import forms
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
 from django.core.serializers.json import DjangoJSONEncoder
-from django.utils.six import with_metaclass, string_types
+from django.db import models
+from django.utils.module_loading import import_string
+from django.utils.six import with_metaclass, string_types, reraise
 
 from wagtail.wagtailcore.rich_text import DbWhitelister, expand_db_html
 from wagtail.utils.widgets import WidgetWithScript
 from wagtail.wagtailcore.blocks import Block, StreamBlock, StreamValue, BlockField
 
 
+class BaseRichTextEditor(object):
```

Comment:
Just discussed with Matthew. He said the following:
 - If the code is for the rich editor interface, it should be in ``wagtailadmin``
 - If the code is for displaying rich text on the frontend, it should be in ``wagtailcore``

So this probably should be moved to ``wagtailadmin``